cmd/simulator/load/loader.go

```diff
@@ -31,6 +35,25 @@ func ExecuteLoader(ctx context.Context, config config.Config) error {
 
 	// Construct the arguments for the load simulator
 	clients := make([]ethclient.Client, 0, len(config.Endpoints))
+	// Extract blockchainIDStr from the clientURI
+	re := regexp.MustCompile(`bc\/(.*)\/`)
+	matches := re.FindStringSubmatch(config.Endpoints[0])
+	if len(matches) < 1 {
+		return fmt.Errorf("failed to get blockchainIDStr from the clientURI %s", config.Endpoints[0])
+	}
+	// Get the last element in matches
+	blockchainIDStr := matches[len(matches)-1]
+	log.Info("Extracted blockchainIDStr from the clientURI", "blockchainIDStr", blockchainIDStr)
+
+	re = regexp.MustCompile(`127.0.0.1:(.*)/ext/bc`)
```

Comment:
can we add the blockchainID as a parameter instead of extracting it from the endpoint here? We should not assume that the blockchainID string will be included in the URI

Reply:
sure

Reply:
done thanks